### DISCUSSION TITLE
Clipboard.SetData behavior (error)

### DISCUSSION THREAD

**Discussion by arumenoy** [twinbasic/twinbasic#1572] [General]
In Visual Basic 6, the following code works fine, but in twinBasic, both lines result in an error with a message saying that there's no object. This software works in VB6.0. I'd appreciate any comments. Thank you.

Clipboard.SetData Form1.Picture1.Image 
Clipboard.SetData Form1.Picture1.Picture

**Comment by wqweto** [comment]
This is current `_Clipboard` interface declaration in TB
```
[ InterfaceId ("33AD4F49-6699-11CF-B70C-00AA0060D393") ]
' FIXME need to match vtable layout to VB6 version
Public Interface _Clipboard Extends stdole.IUnknown        ' FIXME needs to suppot IDispatch
    Sub Clear()
    ' GetData           FIXME missing
    ' SetData           FIXME missing
    Function GetText([ TypeHint (ClipboardConstants) ] Optional ByVal Format As Variant) As String
    Sub SetText(ByVal Str As String, [ TypeHint (ClipboardConstants) ] Optional ByVal Format As Variant)
    Function GetFormat([ TypeHint (ClipboardConstants) ] ByVal Format As Integer) As Boolean
End Interface
```
Apparently `GetData` and `SetData` methods are not implemented yet so there is some more work planned here.

**Comment by WaynePhillipsEA** [comment]
Indeed.  It shouldn't be too much work to get this added, since we already support images etc in the OLE drag/drop DataObject.  I'll bump it up the list.

**Comment by arumenoy** [comment]
Thank you for your comment. However, it's strange. With OLE drag/drop, even in the latest version, I receive an error message during compilation (before execution) and removing the entire function makes it work. I was wondering if it wasn't implemented yet... But in the original VB6, it works without errors.

**Reply by WaynePhillipsEA** [reply]
Could you open a separate issue for that please, including the code that demonstrates the compilation error.

**Reply by arumenoy** [reply]
![OLE_Error](https://github.com/twinbasic/twinbasic/assets/135419894/cf3c7f49-c551-4711-a7f1-4eacaeee0135)
I'm encountering an error with the function itself. Is there anything I need to add? I'm using Visual Basic 6.0 (VB6), and the source code is unchanged and working. Since VB6 defines the function, I believe it's unlikely that there's a "typo" in the code. Any help would be appreciated.

**Reply by wqweto** [reply]
@arumenoy Keep in mind that TB is BETA software and the features you are having troubles with are unimplemented yet.

There is nothing you (or anybody else) can do at the moment except wait for these to get complete.

**Reply by arumenoy** [reply]
>we already support images etc in the OLE drag/drop DataObject.
Sorry. I may have misunderstood something big in this comment.

**Reply by WaynePhillipsEA** [reply]
Interesting.  I've just checked, and OLEDragDrop is generally working fine, and from what I can see in the screenshot, the procedure signature looks OK.  I can only think that the `DataObject` type (Data parameter) is somehow referring to a different version of `DataObject`, perhaps from the real VBRUN (vb6) type library.  

If you put the cursor on the `DataObject` symbol, and press F12.  Then take a screenshot of where it takes you.

**Reply by WaynePhillipsEA** [reply]
There is an open issue for the OLEDragDrop issue, see https://github.com/twinbasic/twinbasic/issues/1544

As discussed over there and mentioned by me above, it is very likely a reference issue.
